### ISSUE TITLE
Proposal: Hide record create and edit controls

### ISSUE THREAD

**Issue by lukepighetti** [pocketbase/pocketbase#5096]
We currently have the ability to `Hide collection create and edit controls`, it would be great if we could also do this for records. That way all data is truly read only from the admin panel.

<img width="1036" alt="Screenshot 2024-06-18 at 8 05 42 AM" src="https://github.com/pocketbase/pocketbase/assets/666539/cbf66bda-1e93-4aac-9da3-88187eb08713">

**Comment by lbensaad** [comment]
@ganigeorgiev I wanted to suggest a new `read only` feature but since this proposal is here, I chose to only comment here instead of creating a new proposal.

What I mean is to add to check box to lock db writes and keep only reads. whenever there is a insert/update/delete request it should fail with an error message "DB is in read only mode", this message can be changed in a text input bellow the checkbox.
And bellow add another checkbox to also lock db for admins or hide controls as @lukepighetti proposed.

**Comment by ganigeorgiev** [comment]
@lbensaad No sorry, I don't see how this would be useful. While we can add knobs for every possible scenario, this will quickly lead to cluttering the UI and I really want to avoid that.
Unless a more prominent use case arise, "read-only" option is out of the scope for now (users can do it manually if they want via the hooks).

In any case as explained in the linked PR the above issue remains open mostly as a remainder to eventually consider it as part of [task#54084615](https://github.com/orgs/pocketbase/projects/2/views/1?filterQuery=hide&pane=issue&itemId=54084615).

**Comment by blackdooo** [comment]
Yeah Great, It would be really useful to have an additional setting to disable record creation and editing directly in the admin panel for a truly read-only setup. This would help enhance data security and prevent any accidental changes. Looking forward to seeing this feature implemented!

**Comment by PSCourtney** [comment]
This would be a useful feature.

Even only adding ability to restrict delete actions for certain superusers would mean the pocketbase UI can become a more user friendly backend. `Hide collection create and edit controls` takes it 90% the way there, `restrict delete` would cap off the last 10% imo.

**Comment by ganigeorgiev** [comment]
It is not exactly what originally was asked here but to prevent accidental data changes on prod an extra "Unlock to save" step has been added in the `develop` branch as part of the new UI and it will be available in the next v0.37.0 release soon _(most likely tomorrow or the beginning of next week)_.

To avoid being too annoying the unlocking currently doesn't ask for extra confirmation (e.g. password) and when the "Hide ..." toggle is enabled it looks like this at the moment:
![record_unlock](https://i.imgur.com/EyBJ62x.png).

A more general "Production mode" will be considered (or the current toggle will be renamed) but that will be done eventually with the ["Stage 2" completion](https://github.com/pocketbase/pocketbase/discussions/7612).

**Comment by PSCourtney** [comment]
This is awesome. 

Thanks so much!
